src/read_write/s2.rs

```diff
@@ -40,8 +47,17 @@ where
     fn write(&mut self, points_batch: &PointsBatch) -> Result<()> {
         let mut batches_by_s2_cell = HashMap::new();
         for (i, pos) in points_batch.position.iter().enumerate() {
+            let radius = pos.magnitude();
+            if radius > EARTH_RADIUS_MAX_M || radius < EARTH_RADIUS_MIN_M {
+                let msg = format!(
+                    "Point ({}, {}, {}) is not a valid ECEF point",
+                    pos.x, pos.y, pos.z
+                );
+                return Err(Error::new(ErrorKind::InvalidInput, msg));
+            }
+            let s2_point = Point::from_coords(pos.x, pos.y, pos.z);
             let s2_cell_batch = batches_by_s2_cell
-                .entry(cell_id(ECEFExt::from(*pos), S2_SPLIT_LEVEL))
+                .entry(CellID::from(s2_point).parent(u64::from(S2_SPLIT_LEVEL)))
```

Comment:
Nit: Make the const `S2_SPLIT_LEVEL` above `u64`, so we don't need the conversion